### PR TITLE
Tenor GIF Integration without Additional Dependencies

### PR DESCRIPTION
**Pull request by ctrl-alt-del**
Tenor GIF Integration without Additional Dependencies

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,13 @@
 # Atlas Android
 
+## 0.4.19
+
+### Major Changes
+  * Update to Layer Android SDK Version 0.23.24
```
Comment (thecombatwombat) [inline CHANGELOG.md]: @ctrl-alt-del Hey sorry about this. We just pulled `0.23.24` due to a bug causing crashes. Could you please downgrade the sdk to `0.23.23`. 

Thanks!

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,5 +1,13 @@
 # Atlas Android
 
+## 0.4.19
+
+### Major Changes
```
Comment (thecombatwombat) [inline CHANGELOG.md]: @ctrl-alt-del Hey sorry about this. We just pulled `0.23.24` due to a bug that was causing crashes. 

Could you please downgrade the sdk to `0.23.23` ?

Comment (smpete) [reply CHANGELOG.md]: I updated the `integration/tenor` branch with develop. This should no longer be necessary.

File: layer-atlas/build.gradle [hunk 3]
```diff
@@ -41,6 +41,7 @@ ext {
     org = "layerhq"
     repo = "Atlas-Android"
     groupFolder = "com/layer/atlas"
+
```
Comment (smpete) [inline layer-atlas/build.gradle]: File should not be changed for just a whitespace change

File: layer-atlas/src/main/java/com/layer/atlas/AtlasMessageComposer.java [hunk 4]
```diff
@@ -114,10 +114,12 @@ public void onClick(View v) {
         mMessageEditText.addTextChangedListener(new TextWatcher() {
             @Override
             public void beforeTextChanged(CharSequence s, int start, int count, int after) {
+
```
Comment (smpete) [inline layer-atlas/src/main/java/com/layer/atlas/AtlasMessageComposer.java]: File should not be changed for just a whitespace change

File: layer-atlas/src/main/java/com/layer/atlas/messagetypes/text/TextCellFactory.java [hunk 5]
```diff
@@ -68,7 +68,6 @@ public TextInfo parseContent(LayerClient layerClient, Message message) {
 
     @Override
     public void bindCellHolder(CellHolder cellHolder, final TextInfo parsed, Message message, CellHolderSpecs specs) {
-
```
Comment (smpete) [inline layer-atlas/src/main/java/com/layer/atlas/messagetypes/text/TextCellFactory.java]: File should not be changed for just a whitespace change

File: layer-atlas/src/main/java/com/layer/atlas/tenor/messagetype/threepartgif/ThreePartGifCellFactory.java [hunk 6]
```diff
@@ -0,0 +1,137 @@
...
+            return;
+        }
+        cellHolder.render(info, message, mOnLoadGifCallback);
+    }
+
+    public void setGifCellHolderOnLoadGifCallback(@NonNull GifCellHolder.OnLoadGifCallback callback) {
+        mOnLoadGifCallback = callback;
+    }
+
+//    @Override
```
Comment (smpete) [inline layer-atlas/src/main/java/com/layer/atlas/tenor/messagetype/threepartgif/ThreePartGifCellFactory.java]: Commented blocks should be removed

File: layer-atlas/src/main/java/com/layer/atlas/AtlasMessageComposer.java (hunk 4 above)
Comment (ctrl-alt-del) [reply layer-atlas/src/main/java/com/layer/atlas/AtlasMessageComposer.java]: Ok

File: layer-atlas/build.gradle (hunk 3 above)
Comment (ctrl-alt-del) [reply layer-atlas/build.gradle]: Ok

**Comment by ctrl-alt-del** [comment]
Hey @thecombatwombat and @smpete , issue on white space caused by auto-indentation is addressed.

**Pull request merged**